### PR TITLE
#620

### LINKED ISSUE #620: New script idea:  Interview Information Sheet
A script to produce an information sheet of high-level case and general information that can be used as a resource when meeting with a client in person.  The top half of the sheet is for the worker to use as a reference and to document any updates that need to be made to the case.  The bottom half of the sheet can be given to the client as a resource.  

For greater efficiency, this script makes use of classes.  We may want to consider whether it makes sense to add the classes to the Functions Library.

A functional draft is ready to pull request.  I am attaching the Word template the script uses.
[Interview Information Sheet.docx](https://github.com/MN-Script-Team/DHS-PRISM-Scripts/files/572461/Interview.Information.Sheet.docx)

### PR DESCRIPTION
**Pull request by levesswe**
Resolves #620 

BLIP: This script produces an information sheet of high-level case and general information that can be used as a resource when meeting with a client in person. The top half of the sheet is for the worker to use as a reference and to document any updates that need to be made to the case. The bottom half of the sheet can be given to the client as a resource.  Counties can make their specific version of the sheet with their own resources.

### CODE REVIEW

File: actions/interview-information-sheet.vbs [hunk 1]
```diff
@@ -0,0 +1,785 @@
+'GATHERING STATS----------------------------------------------------------------------------------------------------
+name_of_script = "ACTIONS - INTERVIEW INFORMATION SHEET.vbs"
```
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: This needs to match the format of the file name exactly. So, "interview-information-sheet.vbs".

File: actions/interview-information-sheet.vbs [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+'Navigating to CAPS
+call navigate_to_PRISM_screen("CAPS")
+
+'Entering case number and transmitting
+EMSetCursor 4, 8
+EMSendKey replace(PRISM_case_number, "-", "")									'Entering the specific case indicated
+EMWriteScreen "d", 3, 29												'Setting the screen as a display action
+transmit															'Transmitting into it
+
+set objNet = CreateObject("WScript.NetWork")
```
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: Wondering why this is here. Needs comments!

File: actions/interview-information-sheet.vbs (hunk 1 above)
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: This needs to match the format of the file name exactly. So, "interview-information-sheet.vbs".

File: actions/interview-information-sheet.vbs [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+username = lcase(objNet.UserName)
+
+
+set info = new doc_info
```
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: Why? Needs comments!

File: actions/interview-information-sheet.vbs [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+
+
+Set objWord = CreateObject("Word.Application")
```
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: Why? Needs comments!

File: actions/interview-information-sheet.vbs [hunk 5]
```diff
@@ -0,0 +1,785 @@
...
+objWord.Visible = True
+
+IF participant = "NCP" Then
```
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: All of this needs TONS more commentary. I am not sure immediately from looking at this part what it's doing or why. You need to explain why it's pulling this information, and what it's doing with it. That's not readily apparent from your code without having PRISM up (we shouldn't assume the audience will never look at the code).

Please beef up comments everywhere throughout the script, especially here.

File: actions/interview-information-sheet.vbs [hunk 6]
```diff
@@ -0,0 +1,785 @@
...
+			.FormFields("cch").Result = FormatCurrency (Ccur(info.cch_amount))
+			.FormFields("cms").Result = FormatCurrency(CCur(info.cms_amount))
+			.FormFields("cmi").Result = FormatCurrency(CCur(info.cmi_amount))
+			.FormFields("ccc").Result = FormatCurrency(CCur(info.ccc_amount))
+			.FormFields("csp").Result = FormatCurrency(CCur(info.csp_amount))
+			.FormFields("arrears_payback").Result = FormatCurrency(info.monthly_non_accrual)
+			.FormFields("ncp_mci").Result = info.ncp_mci	
+	End With
+ELSE
+	set objDoc = objWord.Documents.Add(word_documents_folder_path & "Interview-Information-Sheet.docx")
```
Comment (theVKC) [inline actions/interview-information-sheet.vbs]: Can we make this file name all lower case, and re-upload the file?

**Comment by theVKC** [comment]
Also, please refrain from saying "resolves" in your pull requests. That follows the old procedure, the new is to keep issues open until testing is complete.

**Comment by levesswe** [comment]
The word document is different from what was originally in my issue.  I removed the Anoka-specific information.  I kept the general resources, though, that I believe apply to all counties.  Feel free to get feedback from people at Hackathon and let me know if they want any changes.  Unless they were really speedy with going through Github notifications, they haven’t seen it.

Wendy LeVesseur
Child Support Officer

Anoka County Government Center
2100 3rd Avenue, Suite 401
Anoka, MN 55303
763-422-7323

**Comment by theVKC** [comment]
I'm actually not at hackathon, home sick. No CS folks are there either. Can you repost the Word doc in the original issue so you can solicit feedback?

**Pull request closed without merging**